src/erlcloud_as.erl

```diff
@@ -412,6 +421,60 @@ detach_instances(InstanceIds, GroupName, ShouldDecrementDesiredCapacity, Config)
     end.
 
 
+describe_lifecycle_hooks(GroupName) ->
+    describe_lifecycle_hooks(GroupName, [], erlcloud_aws:default_config()).
+
+describe_lifecycle_hooks(GroupName, LifecycleHookNames) when is_list(LifecycleHookNames) ->
+    describe_lifecycle_hooks(GroupName, LifecycleHookNames, erlcloud_aws:default_config());
+describe_lifecycle_hooks(GroupName, Config) ->
+    describe_lifecycle_hooks(GroupName, [], Config).
+
+-spec describe_lifecycle_hooks(string(), list(string()), aws_config()) -> {ok, list(aws_autoscaling_lifecycle_hook())} | {error, term()}.
+describe_lifecycle_hooks(GroupName, LifecycleHookNames, Config) ->
+    Params = [{"AutoScalingGroupName", GroupName} | member_params("LifecycleHookNames.member.", LifecycleHookNames)],
+    case as_query(Config, "DescribeLifecycleHooks", Params, ?API_VERSION) of
+        {ok, Doc} ->
+            LifecycleHooks = xmerl_xpath:string(?DESCRIBE_LIFECYCLE_HOOKS_PATH, Doc),            
+            Hooks = [extract_lifecycle_hook(H) || H <- LifecycleHooks],
+            {ok, Hooks};
+        {error, Reason} ->
+            {error, Reason}
+    end.
+
+extract_lifecycle_hook(H) -> 
+    #aws_autoscaling_lifecycle_hook{
+          group_name = erlcloud_xml:get_text("AutoScalingGroupName", H),
+          lifecycle_hook_name = erlcloud_xml:get_text("LifecycleHookName", H),
+          global_timeout = erlcloud_xml:get_integer("GlobalTimeout", H),
+          heartbeat_timeout = erlcloud_xml:get_integer("HeartbeatTimeout", H),
+          default_result = erlcloud_xml:get_text("DefaultResult", H),
+          lifecycle_transition = erlcloud_xml:get_text("LifecycleTransition", H)}.
+
+
+-spec complete_lifecycle_action(string(), string(), string(), {instance_id | token, string()}) -> {ok, string()} | {error, term()}.
+complete_lifecycle_action(GroupName, LifecycleActionResult, LifecycleHookName, InstanceIdOrLifecycleActionToken) ->
+    complete_lifecycle_action(GroupName, LifecycleActionResult, LifecycleHookName, InstanceIdOrLifecycleActionToken, erlcloud_aws:default_config()).
+
+-spec complete_lifecycle_action(string(), string(), string(), {instance_id | token, string()}, aws_config()) -> {ok, term()} | {error, term()}.
```

Comment:
Suppose the return spec should be the same `{ok, string()} | {error, term()}` as for the `complete_lifecycle_action()` above